src/app/components/homeMain/charts/volumeChart.tsx

```diff
@@ -0,0 +1,181 @@
+
+      <div className="flex w-full flex-grow items-center justify-center">
+        <ResponsiveContainer width="100%" height="100%">
+          <BarChart
+            width={500}
+            height={400}
+            data={data}
+            margin={{
+              right: 0,
+              left: 0,
+            }}
+          >
+            <XAxis
+              dataKey="date"
+              tickLine={false}
+              tick={{ fontSize: 12 }}
+              minTickGap={30}
+              tickMargin={0}
+              axisLine={false}
+              interval={"preserveStartEnd"}
+              allowDataOverflow
+            />
+            <YAxis hide={true} scale="log" domain={["auto", "auto"]} />
+            <Tooltip
+              content={CustomTooltip as any}
+              cursor={{
+                stroke: "#B374F",
+                strokeWidth: 3,
+              }}
+            />
+            <defs>
+              <linearGradient id="colorBv" x1="0" y1="0" x2="0" y2="1">
+                <stop offset="5%" stopColor="#9D62D9" stopOpacity={1} />
+                <stop offset="95%" stopColor="#B374F2" stopOpacity={0.3} />
+              </linearGradient>
+              <linearGradient id="colorUv" x1="0" y1="0" x2="0" y2="1">
+                <stop offset="5%" stopColor="#7474F2" stopOpacity={0.8} />
+                <stop offset="95%" stopColor="#7474F2" stopOpacity={0} />
+              </linearGradient>
+            </defs>
+            {data && data.length > 0 ? (
+              !data[0].volume2 ? (
+                <Bar
+                  // I may need the commented out properties in the future for styling
+                  //   type="monotone"
+                  dataKey="volume1"
+                  //   activeBar={{ stroke: "red", strokeWidth: 2 }}
+                  //   fillOpacity={1}
+                  //   activeBar={<Dot r={10} fill="white" stroke="black" />}
+                  fill="url(#colorBv)"
+                  //   activeDot={{ stroke: "#8884d8", strokeWidth: 1, fill: "white" }}
+                />
+              ) : (
+                <>
+                  <Bar
+                    //   type="monotone"
+                    dataKey="volume1"
+                    //   activeBar={{ stroke: "red", strokeWidth: 2 }}
+                    //   fillOpacity={1}
+                    //   activeBar={<Dot r={10} fill="white" stroke="black" />}
+                    fill="url(#colorUv)"
+                    //   activeDot={{ stroke: "#8884d8", strokeWidth: 1, fill: "white" }}
+                  />
+                  <Bar
+                    //   type="monotone"
+                    dataKey="volume2"
+                    //   activeBar={{ stroke: "red", strokeWidth: 2 }}
+                    //   fillOpacity={1}
+                    //   activeBar={<Dot r={10} fill="white" stroke="black" />}
+                    fill="url(#colorBv)"
+                    //   activeDot={{ stroke: "#8884d8", strokeWidth: 1, fill: "white" }}
```

Comment:
delete all these comments, unless they have a purpose, if the have a purpose leave a comment